myscript.js

```diff
@@ -0,0 +1,21 @@
+
+var dataArray = [1, 8, 7, [ 4, 65, 1, [ 2, 7 ], 45, [ 6, 4], 6 ], 4, 8]
+
+var jenyaFlatternArray = function (Array) {
+    var flattering = function (mlArray) {
+        for (var i = 0; i < mlArray.length; i++) {
+            var check = typeof mlArray[i];
```

Comment:
Concerning additional local variable `check`, it just didn't work in jsfiddle when I wrote `if (typeof mlArray[i] === "object")`. I couldn't find out the reason,  so I used that needless local variable.
Currently, I changed it as you wrote and it works just fine.

I've added a local variable `var mlArrayI = mlArray[i];`. 
I suppose it should be done so because of the rule of code writing called DRY ('don't repeat yourself'), but can You please explain me more detailed what is the advantage to have another line of code and another one local variable? 
I may suggest it is because it is easier to make a "link" for the computer to an exact place in memory at the address of `mlArray[i]` and later accessing it directly instead of "calculating" its place each time. But it seems weird for me as `mlArray[i]' looks *direct* enough. 
It will be nice to have a short talk with you about this concern.